### PR TITLE
Modification to plot_time_slice 

### PR DESCRIPTION
**Pull request by davidfee5**
These changes modify the view of the RTM stack "slice" in plot_time_slice, including adding an option to plot the peak stack function as a subplot. This is a nice way to view the results in both space and time.

### CODE REVIEW

File: rtm/plotting.py [hunk 1]
```diff
@@ -177,6 +187,21 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
...
+    cbar = fig.colorbar(sm, cax=cbaxes, label='Stack amplitude')
+    cbar.solids.set_alpha(1)
+
+    if plot_peak:
+        s_peak = S.max(axis=(1, 2)).data
+
+        ax1.plot(S.time, s_peak, 'k-')
+        ax1.set_xlim(S.time[0].data, S.time[-1].data)
+        ax1.set_aspect('auto')
+        ax1.set_xlabel('UTC Time')
```
Comment (liamtoney) [inline rtm/plotting.py:202]:
```suggestion
        ax1.set_xlabel('UTC time (HH:MM:SS)')
```

File: rtm/plotting.py [hunk 2]
```diff
@@ -100,7 +114,7 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
         ax.set_xlabel('UTM Easting (m)')
```
Comment (liamtoney) [inline rtm/plotting.py:114]:
```suggestion
        ax.set_xlabel('UTM easting (m)')
```

File: rtm/plotting.py [hunk 3]
```diff
@@ -100,7 +114,7 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
...
         ax.set_ylabel('UTM Northing (m)')
```
Comment (liamtoney) [inline rtm/plotting.py:115]:
```suggestion
        ax.set_ylabel('UTM northing (m)')
```

File: rtm/plotting.py [hunk 4]
```diff
@@ -177,6 +187,21 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
...
+    cbar.solids.set_alpha(1)
+
+    if plot_peak:
+        s_peak = S.max(axis=(1, 2)).data
+
+        ax1.plot(S.time, s_peak, 'k-')
+        ax1.set_xlim(S.time[0].data, S.time[-1].data)
+        ax1.set_aspect('auto')
+        ax1.set_xlabel('UTC Time')
+        ax1.set_ylabel('Peak Stack Amplitude')
```
Comment (liamtoney) [inline rtm/plotting.py:203]: I think this makes it a little more clear?
```suggestion
        ax1.set_ylabel('Stack amplitude at spatial stack max')
```

File: rtm/plotting.py (hunk 1 above)
Comment (liamtoney) [reply rtm/plotting.py]: Actually, format is not always `HH:MM:SS`, so scratch this

File: rtm/plotting.py [hunk 5]
```diff
@@ -421,8 +421,8 @@ def plot_stack_peak(S, plot_max=False, ax=None):
                        color='red', edgecolor='black', s=150, zorder=5)
 
     ax.set_xlim(S.time[0].data, S.time[-1].data)
-    ax.set_xlabel('UTC Time')
-    ax.set_ylabel('Peak Stack Amplitude')
+    ax.set_xlabel('UTC time')
+    ax.set_ylabel('Stack amplitude at spatial stack max')
```
Comment (davidfee5) [inline rtm/plotting.py]: This results in a really long label that extends well beyond the axes limit, so I reverted to something shorter. Can add more info in the text or caption.

Comment (liamtoney) [reply rtm/plotting.py]: :+1:

**Comment by davidfee5** [comment]
All looks good to me. Ready to merge when you are!

**Comment by liamtoney** [comment]
Some late-breaking commits here:

* Make spiffy contours
* Force scalebar length to be a multiple of 100
* Mask areas outside of DEM extent for projected grids
* Increase visibility of markers and labels

All aesthetic. They're modular, we can revert any/all that we don't agree on! The differences for local are shown below. Fewer differences for regional (just the marker color change). Old on top, new on bottom:

old

new

File: rtm/plotting.py [hunk 6]
```diff
@@ -151,8 +151,9 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
         plot_transform = ax.transData
 
         # Mask areas outside of DEM extent
-        slice.data[np.isnan(dem.data)] = np.nan
-
+        dem_slice = dem.sel(x=slice.x, y=slice.y, method='nearest')
+        slice.data[np.isnan(dem_slice.data)]
```
Comment (liamtoney) [inline rtm/plotting.py:155]: That'll work better! 😉 
```suggestion
        slice.data[np.isnan(dem_slice.data)] = np.nan
```

File: rtm/plotting.py [hunk 7]
```diff
@@ -151,8 +151,9 @@ def plot_time_slice(S, processed_st, time_slice=None, label_stations=True,
         plot_transform = ax.transData
 
         # Mask areas outside of DEM extent
-        slice.data[np.isnan(dem.data)] = np.nan
-
+        dem_slice = dem.sel(x=slice.x, y=slice.y, method='nearest')
```
Comment (liamtoney) [inline rtm/plotting.py:154]: Just adding a comment, change if I'm not interpreting this right
```suggestion
        dem_slice = dem.sel(x=slice.x, y=slice.y, method='nearest')  # Select subset of DEM that slice occupies
```

**Comment by davidfee5** [comment]
OK! Thanks for the help. Think we are good to go here so going to merge.

**Pull request merged**